### PR TITLE
remove need to clone core repo

### PR DESCRIPTION
**Pull request by codeboten**
# Description

Companion PR to https://github.com/open-telemetry/opentelemetry-python/pull/2108, this removes the need to clone the core repo in order to run tox. It uses pip instead.

NOTE: This change also removes lint ignore rules for the `opentelemetry-python-core` subdirectory. This means after this PR, contributors should remove that folder from their local environment.

### CODE REVIEW

**Review by lonewolf3739 — APPROVED** [review]

**Review by owais — APPROVED** [review]

**Comment by owais** [comment]
Generate for bootstrap needs to be updated. I guess we should move it to core and use something [like this](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/516) as source of truth for generate

I was planning to come back and finish the meta package PR next week but if it helps, feel free to take it over. https://github.com/open-telemetry/opentelemetry-python-contrib/pull/516

**Comment by codeboten** [comment]
@owais yah that's the last thing i need to fix for this PR to be done. will take a look sometimes this week.

**Comment by lzchen** [comment]
@owais 
Which files for bootstrap needs to be updated exactly? (or not, if we are changing it)

**Review by lzchen — APPROVED** [review]

**Comment by owais** [comment]
@lzchen this file: https://github.com/open-telemetry/opentelemetry-python/blob/main/opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py

@codeboten I think I'll something that might help you with this today.

**Comment by lzchen** [comment]
@owais 
The `gen` file has to be updated every release. Is that related to this PR though?

**Comment by lzchen** [comment]
@owais 
Oh I guess `opentelemetry-python-core` needs to be checked out to run `eachdist.py` for generating?

**Comment by owais** [comment]
@codeboten  see if this helps: https://github.com/open-telemetry/opentelemetry-python-contrib/pull/681/files#diff-02fe7adf3ca73fd04ef2811691de61fbc8ae90a52016caa844922c4b811d9e8dR29

You can install the egg from git like other packages as a dev dependency and just copy over the `__init__.py` file from the metapackage.

**Comment by owais** [comment]
> Oh I guess opentelemetry-python-core needs to be checked out to run eachdist.py for generating?

Right. That's how it works today but with the changes @codeboten is making, it shouldn't be necessary going forward.

**Comment by codeboten** [comment]
@owais please take another look. I've updated `generate_instrumentation_bootstrap.py` to:

- pull down `bootstrap_gen.py` from the core repo via `requests` and store it in a temp folder
- generate the new `bootstrap_gen.py` and compare the two files
- if they differ, the script returns an error, which will cause CI to fail, rather than running `git diff` as was done before

Let me know what you think.

**Comment by owais** [comment]
IIUR, this will tell the author that bootstrap_gen in core needs an update and then the author will have to go and create a PR in core by manually copying/generating the file. Is that right?

**Comment by owais** [comment]
It would be great if a bot could create PR in core repo every time the file changed in contrib main but I'm working on a proposal related to distros that might help avoid this completely.

**Comment by codeboten** [comment]
> IIUR, this will tell the author that bootstrap_gen in core needs an update and then the author will have to go and create a PR in core by manually copying/generating the file. Is that right?

@owais yup that's correct. it's not great, but it seems the benefits of removing the core repo from regular dev workflow outweighs the burden to manually copy this file when necessary. I like the idea of removing the need for this altogether though, that sounds promising! Thoughts?

**Comment by owais** [comment]
Definitely! I wasn't against it. Just trying to verify if I understood the new flow properly. We should just document the workflow here so outside contributors know what they are supposed to do to fix the CI failure.

File: scripts/generate_instrumentation_bootstrap.py [hunk 1]
```diff
@@ -50,16 +52,17 @@
...
-    "opentelemetry-python-core",
-    "opentelemetry-instrumentation",
-    "src",
-    "opentelemetry",
-    "instrumentation",
-    "bootstrap_gen.py",
+current_path = os.path.join(tmpdir.name, "current.py",)
+
+core_repo = os.getenv("CORE_REPO_SHA", "main")
+url = "https://raw.githubusercontent.com/open-telemetry/opentelemetry-python/{}/opentelemetry-instrumentation/src/opentelemetry/instrumentation/bootstrap_gen.py".format(
```
Comment (ocelotl) [inline scripts/generate_instrumentation_bootstrap.py:61]: Better to make this a f-string

**Review by ocelotl — APPROVED** [review]

**Comment by ocelotl** [comment]
@codeboten, please check `lint`.

**Comment by codeboten** [comment]
@ocelotl fixed the lint errors

**Pull request merged**